ipatests/test_integration/test_installation.py

```diff
@@ -1119,6 +1129,110 @@ def run_request(url, expected_stdout=None, expected_stderr=None):
+
+        https://pagure.io/freeipa/issue/2692
+        """
+        original_hostname = self.master.hostname
+        new_hostname = 'new.' + original_hostname
+        # New hostname is added into /etc/hosts as the installer
+        # is looking for it for resolution. Without it, an installation
+        # fails with `Unable to resolve host name, check /etc/hosts or DNS
+        # name resolution`.
+        hosts = self.master.get_file_contents(paths.HOSTS, encoding='utf-8')
+        new_hosts = hosts.replace(original_hostname, new_hostname)
+        self.master.put_file_contents(paths.HOSTS, new_hosts)
+        try:
+            cmd = ['ipa-server-install', '--hostname', new_hostname]
+            with self.master.spawn_expect(cmd) as e:
+                e.expect_exact('Do you want to configure integrated '
+                               'DNS (BIND)? [no]: ')
+                e.sendline('no')
+                e.expect_exact('Please confirm the domain name [{}]: '.format(
+                    original_hostname  # DN is computed from new hostname
+                ))
+                e.sendline(self.master.domain.name)
+                e.expect_exact('Please provide a realm name [{}]: '.format(
+                    self.master.domain.realm
+                ))
+                e.sendline(self.master.domain.realm)
+                e.expect_exact('Directory Manager password: ')
+                e.sendline(self.master.config.dirman_password)
+                e.expect_exact('Password (confirm): ')
+                e.sendline(self.master.config.dirman_password)
+                e.expect_exact('IPA admin password: ')
+                e.sendline(self.master.config.admin_password)
+                e.expect_exact('Password (confirm): ')
+                e.sendline(self.master.config.admin_password)
+                e.expect_exact('Do you want to configure chrony with '
+                               'NTP server or pool address? [no]: ')
+                e.sendline('no')
+                e.expect_exact('Continue to configure the system '
+                               'with these values? [no]: ')
+                e.sendline('yes')
+                e.expect_exit(ignore_remaining_output=True, timeout=720)
+            hostname = self.master.run_command([
+                'hostname', '-f']).stdout_text.strip()
+            assert hostname == new_hostname
+        finally:
+            # no need to restore the hostname as the installer
+            # does it during uninstallation
+            self.master.put_file_contents(paths.HOSTS, hosts)
+
+    def test_ad_subpackage_dependency(self, server_cleanup):
+        """
+        Test if the installer is not dependant on trust-ad package and
+        succeeds even when AD subpackage is not installed
+
+        https://pagure.io/freeipa/issue/4011
+        """
+        package_name = '*ipa-server-trust-ad'
+        reinstall = False
+        if tasks.is_package_installed(self.master, package_name):
```

Comment:
This internaly calls `'rpm' '-q' '*ipa-server-trust-ad'`. Unlike `dnf` the `rpm` does not expand wildcards and this check will allways return False